optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/GizmoMemberAccessorEntityEnhancer.java

```diff
@@ -482,6 +486,13 @@ public static String generateKieRuntimeBuilder(ClassOutput classOutput, ScoreDir
                                 KieBaseExtractor.class),
                         methodCreator.getMethodParam(0), kieBaseExtractor);
 
+                transformers.produce(new BytecodeTransformerBuildItem(config.getSolutionClass().getName(),
+                        (className, classVisitor) -> new OptaPlannerDroolsReactiveClassVisitor(config.getSolutionClass(),
+                                classVisitor)));
+                for (Class<?> entityClass : config.getEntityClassList()) {
+                    transformers.produce(new BytecodeTransformerBuildItem(entityClass.getName(),
+                            (className, classVisitor) -> new OptaPlannerDroolsReactiveClassVisitor(entityClass, classVisitor)));
+                }
```

Comment:
I'm afraid this is not enough. Property reactivity could be declared on any problem fact as well.
I think we should just allow prop reactive, no questions asked. (Just like it is done on DRL outside of Quarkus.)

Reply:
(can problem facts change during a solver run? My understanding is the solver is restarted after a PFC, in which case the problem fact is removed and re-inserted (vs notifying via changed property like planning entity)

Reply:
Is it worth it? I am not sure if we can do this in a foolproof way. Literally any class on the classpath could be configured to be property reactive - you can't scan it all, can you?

Reply:
OptaPlanner only touches Planning Entities in the DRL. And we know the list of planning entities classes. Additional, classes are configured to be property reactive via annotations, which we can (and do) read.

Reply:
Ok, so let me put it another way - let's write a test that shows that prop reactivity works as expected.
That means that there should be a DRL that has some rules fired (and some not) based on the fields of objects that changed.

I really do not think we should mess with this. It is not our business - prop reactive is a Drools feature, not OP feature. But if we decide to mess with this, let's make sure we do it correctly.